### DISCUSSION TITLE
Finnish language support

### DISCUSSION THREAD

**Discussion by akx** [meilisearch/product#630] [Feedback & Feature Proposal]
# Finnish support

Finnish is a language that heavily uses [inflection](https://en.wikipedia.org/wiki/Inflection) and [agglutination](https://en.wikipedia.org/wiki/Agglutination), as well as compound words (https://github.com/meilisearch/meilisearch/discussions/2709 for prior discussion around compound words re Norwegian and other Nordic languages – however it's important to note that Finnish is entirely unrelated to the other languages spoken in the Nordics).

This tends to make Finnish a bit of a challenge for search; for instance, a document might contain the agglutinated inflected word "kävelykaduille" [(on)to pedestrian streets]; one would expect to find it with the search "kävely katu" (uninflected, compound word split).

## Prior art

Some prior art/software I know of around this space:

* https://voikko.puimula.org/ – free linguistic software for Finnish
* https://github.com/EvidentSolutions/raudikko – a Java port of Voikko (GPL license)
* https://github.com/EvidentSolutions/elasticsearch-analysis-raudikko/ – ElasticSearch analysis plugin using Raudikko (GPL license)

**Comment by ManyTheFish** [comment]
Hello @akx,
Thank you for your detailed proposal and sorry for the response time, I'll try to find the time to study these resources as soon as possible in order to implement Finnish support in the Meilisearch tokenizer.
To be honest with you, I can't promise a release date for the Finnish support, 😞
However, if you know a bit about Rust development or are interested to try it, our tokenizer repository, [Charabia](https://github.com/meilisearch/charabia), is a good way to contribute easily to a Rust project, on top of it, if you need help, I can guide you during your implementation.
We already have helper structures that take a word dictionary in parameters to segment a text.

Thank you for your feedback and sorry again for the time!
